### PR TITLE
feat: add filter option for `select`

### PR DESCRIPTION
**Pull request by alirezaarzehgar**
### Description
We have Filter option for multiselect printer but this feature is not available for select printer.
This PR implemented filter option for select printer.

- [ ] Bug Fix
- [x] New Feature
- [ ] Documentation
- [ ] Other

### To-Do Checklist
- [x] I tested my changes
- [x] I have commented every method that I created/changed
- [ ] I updated the examples to fit with my changes
- [x] I have added tests for my newly created methods

### CODE REVIEW

**Comment by MarvinJWendt** [comment]
Hi @alirezaarzehgar, thanks for the PR! Could you please merge `master` into this PR, and solve the conflicts? This should also fix the failed check, as we switched something in our CI System that was faulty.

File: interactive_select_printer.go [hunk 1]
```diff
@@ -78,6 +80,12 @@ func (p InteractiveSelectPrinter) WithOnInterruptFunc(exitFunc func()) *Interact
 	return &p
 }
 
+// WithFilter sets the Filter option
+func (p InteractiveSelectPrinter) WithFilter(filter bool) *InteractiveSelectPrinter {
+	p.Filter = filter
+	return &p
+}
```
Comment (MarvinJWendt) [inline interactive_select_printer.go:87]:
```suggestion
func (p InteractiveSelectPrinter) WithFilter(b ...bool) *InteractiveSelectPrinter {
	p.Filter = internal.WithBoolean(b)
	return &p
}
```

**Review by MarvinJWendt — CHANGES_REQUESTED** [review]
Hi, looks good to me, only one thing that I noted below. You can use the web editor to merge the suggestion, if you want to.

**Comment by alirezaarzehgar** [comment]
> Hi, looks good to me, only one thing that I noted below. You can use the web editor to merge the suggestion, if you want to.

Hello. Done. I accepted your recommendation and resolve conflicts for now.

**Review by MarvinJWendt — APPROVED** [review]
LGTM! Thanks for contributing 🚀

**Pull request merged**